samcli/lib/providers/sam_function_provider.py

```diff
@@ -387,8 +362,25 @@ def _parse_layer_info(
             # In the list of layers that is defined within a template, you can reference a LayerVersion resource.
             # When running locally, we need to follow that Ref so we can extract the local path to the layer code.
             if isinstance(layer, dict) and layer.get("Ref"):
-                layer_logical_id = cast(str, layer.get("Ref"))
-                layer_resource = stack.resources.get(layer_logical_id)
+                # Here layer refers to another resources.
+                # in a multi stack system, the layer might locate in the current stack or passed down from parents.
+                # for example, {"Ref": "layerLogicalID"} refers to the "layerLogicalID" in the current stack.
+                # {"Ref": {"CrossStackRef": ["..", "layerLogicalID"]}
+                #   refers to the "layerLogicalID" in the parent stack.
+                # {"Ref": {"CrossStackRef": ["..", {"CrossStackRef": ["..", "layerLogicalID"]})}
```

Comment:
We carefully clear this `CrossStackRef` from the template. 

And what happens if there are more than one level between the layer definition vs the function. Like `RootStack (LayerDefinition) -> ChildStack -> GrandChildStack (Function which uses LayerDefinition from RootStack)`?

Reply:
> We should carefully clear this CrossStackRef from the template.

Yes, but this does not affect the output template because it is only affected by `builder.update_template()`(only update codeuri and other path properties)  and `template.move_template()` (only move the file). 

`get_translation()` is only used in `IntrinsicResolver`, which is only used in `SamBaseProvider.get_template()`, which 

> And what happens if there are more than one level between the layer definition vs the function.

this is the last example. I tested this with

```yaml

# template.yaml
Resources:
  MyLayerVersion:
    Type: AWS::Serverless::LayerVersion
    Properties:
      LayerName: MyLayer
      Description: Layer description
      ContentUri: MyLayerVersion
      CompatibleRuntimes:
        - python3.8

  App:
    Type: AWS::Serverless::Application
    Properties:
      Location: ./child.yaml
      Parameters:
        Layer: !Ref MyLayerVersion

# child.yaml
Parameters:
  Layer:
    Type: String

Resources:
  Function:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: function
      Handler: app.lambda_handler
      Runtime: python3.8
      Layers:
        - !Ref Layer
  
  App:
    Type: AWS::Serverless::Application
    Properties:
      Location: ./child2.yaml
      Parameters:
        Layer: !Ref Layer

# child2.yaml
Parameters:
  Layer:
    Type: String

Resources:
  Function2:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: function
      Handler: app.lambda_handler
      Runtime: python3.8
      Layers:
        - !Ref Layer

```